librz/include/rz_type.h

```diff
@@ -175,6 +175,44 @@ typedef struct rz_type_path_t {
 	char *path;
 } RzTypePath;
 
+/**
+ * \brief Type Conditions
+ */
+typedef enum {
+	RZ_TYPE_COND_AL = 0, ///< Always executed (no condition)
+	RZ_TYPE_COND_EQ, ///< Equal
+	RZ_TYPE_COND_NE, ///< Not equal
+	RZ_TYPE_COND_GE, ///< Greater or equal
+	RZ_TYPE_COND_GT, ///< Greater than
+	RZ_TYPE_COND_LE, ///< Less or equal
+	RZ_TYPE_COND_LT, ///< Less than
+	RZ_TYPE_COND_NV, ///< Never executed             must be a nop? :D
+	RZ_TYPE_COND_HS, ///< Carry set                  >, ==, or unordered
+	RZ_TYPE_COND_LO, ///< Carry clear                Less than
+	RZ_TYPE_COND_MI, ///< Minus, negative            Less than
+	RZ_TYPE_COND_PL, ///< Plus, positive or zero     >, ==, or unordered
+	RZ_TYPE_COND_VS, ///< Overflow                   Unordered
+	RZ_TYPE_COND_VC, ///< No overflow                Not unordered
+	RZ_TYPE_COND_HI, ///< Unsigned higher            Greater than, or unordered
+	RZ_TYPE_COND_LS ///< Unsigned lower or same     Less than or equal
+} RzTypeCond;
+
+/**
+ * \brief Variable constrained by the type conditions
+ */
+typedef struct rz_type_var_constraint_t {
+	RzTypeCond cond;
+	ut64 val;
+} RzTypeVarConstraint;
+
+/**
+ * \brief Wrapper of RzType and RzTypeVarConstraint
+ */
+typedef struct rz_constrained_type_t {
+	RzVector /*<RzTypeVarConstraint>*/ constraints;
+	RzType *type;
+} RzConstrainedType;
```

Comment:
I would remove the `type` member from here and rename the struct to `RzTypeConstraints` or similar, ...

Reply:
Then it would be just like before, both `constraint vector` and  `type` stay independent in `RzAnalysisVar`?

Reply:
Yes, lets do then this way - separate type and constraints.